orchagent/portsorch.cpp

```diff
@@ -1860,6 +1887,24 @@ void PortsOrch::initPortSupportedSpeeds(const std::string& alias, sai_object_id_
     m_portStateTable.set(alias, v);
 }
 
+void PortsOrch::initPortCapabilityLinkTraining(const Port &port)
+{
+    sai_status_t status;
+    sai_attribute_t attr;
+
+    attr.id = SAI_PORT_ATTR_SUPPORTED_LINK_TRAINING_MODE;
+    status = sai_port_api->set_port_attribute(port.m_port_id, &attr);
```

Comment:
I don't get this logic. This is not the way to query and understand capabilities. We have separate API sai_query_attribute_capability. Please use that API.

Reply:
As discussed in https://github.com/Azure/sonic-swss/pull/2215, sai_query_attribute_capability does not support per-port basis capability query, while the AN/LT may only be available on certain ports on the switch ASIC

Reply:
Thanks, will fix it

Reply:
Still not fixed